### PR TITLE
Use ETL to fetch data and calculate accuracy from duckDB

### LINKED ISSUE #854: Update Accuracy API to use ETL + DuckDB
On the ETL + DuckDB flow the Accuracy data endpoint that we use to serve accuracy data to pdr-web trough an API it's broken because of some slot fetching updates. Before merging the ETL + DuckDB we should make sure that the Accuracy endpoint is also working.
While on this we should also use the data from the ETL to calculate the Accuracy instead of fetching and storing the data separately.

Dod:
- [x] Fix Accuracy endpoint so it fetches and serves data correctly with the new subgraph fetching changes
- [ ] Integrate ETL + DuckDB so the accuracy endpoint uses ETL to fetch and store data

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #854.

### CODE REVIEW

File: pdr_backend/accuracy/app.py [hunk 1]
```diff
@@ -77,13 +79,18 @@ def process_single_slot(
         int(slot.ID.split("-")[1])
     )  # Using dot notation for attribute access
 
+    today = 0
```
Comment (calina-c) [inline pdr_backend/accuracy/app.py:82]: these variables are named like they represent times, but they are ints. What are they supposed to hold?

File: pdr_backend/accuracy/app.py [hunk 2]
```diff
@@ -239,12 +248,31 @@ def calculate_timeframe_timestamps(
         # else timedelta(days=1)
     )
     start_ts = UnixTimeS(int((datetime.utcnow() - time_delta).timestamp()))
-
     return start_ts, end_ts
 
 
 @enforce_types
-def save_statistics_to_file():
+def fetch_statistics_using_ETL():
```
Comment (calina-c) [inline pdr_backend/accuracy/app.py:255]: shouldn't the ppss be customisable?

File: pdr_backend/accuracy/app.py [hunk 3]
```diff
@@ -239,12 +248,31 @@ def calculate_timeframe_timestamps(
...
+    )
+
+    gql_data_factory = GQLDataFactory(ppss)
+    etl = ETL(ppss, gql_data_factory)
+    while True:
+        etl.do_etl()
+        threading.Event().wait(300)  # Wait for 5 minutes (300 seconds)
+
+
+def filter_func(spe):
```
Comment (calina-c) [inline pdr_backend/accuracy/app.py:270]: please document and type hint/enforce

File: pdr_backend/accuracy/app.py [hunk 4]
```diff
@@ -342,12 +352,37 @@ def serve_statistics_from_file():
...
+    end_ts = UnixTimeS(int(datetime.utcnow().timestamp()))
     try:
-        with open(JSON_FILE_PATH, "r") as f:
-            data = json.load(f)
-            response = jsonify(data)
-            response.headers.add("Access-Control-Allow-Origin", "*")  # Allow any origin
-            return response
+        db_conn = PersistentDataStore("./lake_data", read_only=True)
+        slots_table = db_conn.query_data(
+            f"""
```
Comment (calina-c) [inline pdr_backend/accuracy/app.py:360]: please align opening and close quotes

File: pdr_backend/accuracy/app.py [hunk 5]
```diff
@@ -342,12 +352,37 @@ def serve_statistics_from_file():
     If the file cannot be read or another error occurs, it returns a 500 Internal Server Error.
     """
 
+    start_ts = UnixTimeS(int((datetime.utcnow() - timedelta(weeks=4)).timestamp()))
```
Comment (calina-c) [inline pdr_backend/accuracy/app.py:355]: for better readability, I would separate the date times. I.e.
four_weeks_ago = datetime.utcnow() - timedelta(weeks=4)
start_ts = UnixTimeS(int(four_weeks_ago.timestamp()))

and same for end

File: pdr_backend/lake/etl.py [hunk 6]
```diff
@@ -168,30 +168,46 @@ def _get_max_timestamp_values_from(
...
+        all_db_tables = PersistentDataStore(
+            self.ppss.lake_ss.lake_dir
+        ).get_table_names()
+
+        queries = []
+
+        for table_name in table_names:
+            table_name_with_type = get_table_name(table_name, table_type)
+            if table_name_with_type not in all_db_tables:
+                print(
```
Comment (calina-c) [inline pdr_backend/lake/etl.py:180]: please replace all print statements with logging

File: pdr_backend/lake/etl.py [hunk 7]
```diff
@@ -218,11 +234,17 @@ def _calc_bronze_start_end_ts(self):
             else datetime.strptime(self.ppss.lake_ss.st_timestr, "%Y-%m-%d_%H:%M")
         )
 
+        fin_timestamp = (
```
Comment (calina-c) [inline pdr_backend/lake/etl.py:237]: this is very confusing. It seems in one case, you assign this variable the result of strptime, and in another case the value of strftime. But those are not the same data type. Should this be a string or a date time?

File: ppss.yaml [hunk 8]
```diff
@@ -5,7 +5,7 @@ lake_ss:
     - binance BTC/USDT ETH/USDT 5m
   #    - binance BTC/USDT ETH/USDT BNB/USDT XRP/USDT ADA/USDT DOGE/USDT SOL/USDT LTC/USDT TRX/USDT DOT/USDT 5m
   #    - kraken BTC/USDT 5m
-  st_timestr: 2023-06-01_00:00 # starting date for data
+  st_timestr: 2024-04-10_00:00 # starting date for data
```
Comment (calina-c) [inline ppss.yaml:8]: should this change be committed?

**Review by calina-c — CHANGES_REQUESTED** [review]

File: pdr_backend/accuracy/app.py (hunk 2 above)
Comment (KatunaNorbert) [reply pdr_backend/accuracy/app.py:255]: I think it's good as it is for now, the scope of the API is quite specific

File: pdr_backend/lake/etl.py (hunk 7 above)
Comment (KatunaNorbert) [reply pdr_backend/lake/etl.py:237]: I guess strptime, some of the canges are from other merges in the base PR

File: ppss.yaml (hunk 8 above)
Comment (KatunaNorbert) [reply ppss.yaml:8]: no, it shouldn't

**Pull request merged**